### ISSUE TITLE
Comments for the Enum draft

### ISSUE THREAD

**Issue by AnyCPU** [ninedraft/enum4go#1]
Hello
I have read https://github.com/ninedraft/enum4go/blob/master/enum_ru.md
TL;DR.
Your draft looks like a C-style union. It is difficult to understand. 

See following example
```
type Fruit enum {
    Unknown
    One
    TheAnswer = 42
    _
    _
    Forty Five
}
```

The `int` type is either default or fallback type. There is no need to write the `iota` after member's name.
An `iota`-ish behavior is by default and it is only for integers. 
If a compilers meets a direct assignment like TheAnswer after that it continues assigning values starting from last direct assignment.
If you want to skip an one value, you can use  the `_` symbol as many as you want.
You should not use different type in an one enum declaration.
The first member of an enum prescribes the storage type of an enum.
The storage type can be any simple type.

Thanks

**Comment by ninedraft** [comment]
Hello and thanks for the comment!

The type that I describe is not equivalent to an union, since it allows not only storing values ​​of different types, but also allows you to uniquely determine the type of value during program execution. I agree that the type_id ordering procedure requires separate discussion (although I believe that the ability to use iota to explicitly assign type id is necessary for some specific cases), but strongly disagree about the constraint on one stored data type. 

The whole idea of ​​this type is in the type safe storage of heterogeneous values in one strongly typed variable.

Of course, this is not my invention, you can find the same mechanisms in other programming languages, where they are called "tagged union" or "sum type".

**Comment by AnyCPU** [comment]
I understand. It also looks like a `variant` type. But I think we have some misleading meanings in the title and content.
